pydis_site/apps/api/viewsets/bot/infraction.py

```diff
@@ -271,3 +272,20 @@ def partial_update_expanded(self, *args, **kwargs) -> Response:
         """
         self.serializer_class = ExpandedInfractionSerializer
         return self.partial_update(*args, **kwargs)
+
+    def create(self, request: HttpRequest, *args, **kwargs) -> Response:
+        """
+        Create an infraction for a target user.
+
+        Called by the Django Rest Framework in response to the corresponding HTTP request.
+        """
+        try:
+            return super().create(request, *args, **kwargs)
+        except IntegrityError:
+            raise ValidationError(
+                {
+                    'non_field_errors': [
+                        'This user already has an active infraction of this type.',
```

Comment:
Is it safe to assume all `IntegrityError`s here will be due to a duplicate infraction? Isn't an `IntegrityError` a fairly generic error?

Reply:
It'll be whenever the relational integrity of the database fails. EG duplicate key, foreign key check or any other check added to the table.

So IG this could also be raised if a foreign key being inserted doesn't exist in the related table.

Reply:
I thought about checking the constraint name here, but (apart from forcing an insert of an ID, which we don't do), we only have a single unique constraint here

Reply:
It's not too troublesome to check the constraint name, right? Do you think it'd be better to do this so this code is "future proof" against future constraints we may add? Or am I being overly cautious?

Reply:
The thing is that the constraint name isn't directly included as an attribute on the error, so we'd need to either just do an "in" check on the string or parse it ourselves, and I didn't quite like that approach...

Reply:
Yeah, screw it. This is fine.